zsoltbalogh/liferay-docker
# SRE-2976 adding modsecurity setup

https://issues.liferay.com/browse/SRE-2976

Main changes:

- Added modsecurity config and CRS ruleset (that's the biggest chunk of this PR) with my tweaking
- Turned off nginx access logs to reduce log-spamming
- Webserver service has now visible logs when using docker-compose
- Modsec can be turned on when using build.sh
- Modsec does not log to stdout, individual transactions can be found in `/var/logs/modsec_audit.log` - only nginx error logs pop up on stdout, with little context

Testing:

- Spin up a new environment (I used 7.4-next branch)
- Navigate to spinner-test.com
- Sign in
- Left menu / Design / Page Templates
- Select Page Templates again, then click on New
- Give a simple name, then click on the Blue + button, then select Content Page Template
- You'll be navigated into an editor. Grab the HTML block from Basic compontents, then pull that into the middle.
- Try to click inside the HTML block that you dropped, it should bring on a HTML editor
- Paste: <script>alert(1)</script>
- Try to save
- An error should pop up on the bottom left side
- Check terminal, there should be some error messages from Modsec

If you're blocked on this process, and something seems loading but does not, probably it got blocked by modsec. We should investigate that, see what rule got triggered, then tone it down/disable it completely.
